app/controllers/api_controller/generic.rb

```diff
@@ -48,7 +48,7 @@ def add_resource(type, _id, data)
               "Resource id or href should not be specified for creating a new #{type} resource"
       end
       subcollections     = cspec[:subcollections]
-      subcollection_data = subcollections.each_with_object({}) do |sc, hash|
+      subcollection_data = Array(subcollections).each_with_object({}) do |sc, hash|
```

Comment:
The tests for tag creation were failing without this, where `subcollections here was returning `nil`. Bug, or am I doing something wrong?